core/AssetManager/UIAssetFetcher/PluginUmdAssetFetcher.php

```diff
@@ -202,7 +203,7 @@ protected function retrieveFileLocations()
             }
 
             if (!$foundChunk) {
-                throw new \Exception("Could not find chunk {$this->requestedChunk}");
+                throw new BadRequestException("Could not find chunk {$this->requestedChunk}", 404);
```

Comment:
Is the exception class checked anywhere? Or can we just pick the `HttpCodeException` for the 404, instead of using a "bad request" with a 404 code?